src/index.js

```diff
@@ -4,15 +4,17 @@ import getConfig from 'next/config';
 
 const { publicRuntimeConfig } = getConfig();
 
-const prefixPath = path => publicRuntimeConfig.assetPrefix.replace(/\/+$/, '') + '/' + (path || '').replace(/^\/+/, '')
+const prefixPath = path =>
+  publicRuntimeConfig.assetPrefix.replace(/\/+$/, '') +
+  '/' +
+  (path || '').replace(/^\/+/, '');
 
 export const prefixURL = url => {
   if (typeof url === 'object') {
-    return { ...url, pathname: prefixPath(url.pathname) };
-  } else {
-    return /^(?:[a-z][a-z0-9+.-]*:|\/\/)/i.test(url) ? url : prefixPath(url);
+    return Object.assign({}, url, { pathname: prefixPath(url.pathname) });
```

Comment:
NOTE: I changed the object spread to an `Object.assign()` to reduce the gzipped size by 200 bytes 👍